packages/block-editor/src/components/responsive-block-control/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+import { Fragment } from '@wordpress/element';
+
+import {
+	ToggleControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ResponsiveBlockControlLabel from './label';
+
+function ResponsiveBlockControl( props ) {
+	const {
+		title,
+		property,
+		toggleLabel,
+		onIsResponsiveChange,
+		renderDefaultControl,
+		renderResponsiveControls,
+		isResponsive = false,
+		defaultLabel = {
+			id: 'all',
+			label: __( 'All' ), /* translators: 'Label. Used to signify a layout property (eg: margin, padding) will apply uniformly to all screensizes.' */
+		},
+		viewports = [
+			{
+				id: 'small',
+				label: __( 'Small screens' ),
+			},
+			{
+				id: 'medium',
+				label: __( 'Medium screens' ),
+			},
+			{
+				id: 'large',
+				label: __( 'Large screens' ),
+			},
+		],
+	} = props;
+
+	if ( ! title || ! property || ! renderDefaultControl ) {
+		return null;
+	}
+
+	/* translators: 'Toggle control label. Should the property be the same across all screen sizes or unique per screen size.'. %s property value for the control (eg: margin, padding...etc) */
+	const toggleControlLabel = toggleLabel || sprintf( __( 'Use the same %s on all screensizes.', ), property );
+
+	/* translators: 'Help text for the responsive mode toggle control.' */
+	const toggleHelpText = __( 'Toggle between using the same value for all screen sizes or using a unique value per screen size.' );
+
+	const defaultControl = renderDefaultControl( <ResponsiveBlockControlLabel property={ property } viewport={ defaultLabel } />, defaultLabel );
+
+	const defaultResponsiveControls = () => {
+		return viewports.map( ( viewport ) => (
+			<Fragment key={ viewport.id }>
+				{ renderDefaultControl( <ResponsiveBlockControlLabel property={ property } viewport={ viewport } />, viewport ) }
+			</Fragment>
+		) );
+	};
+
+	return (
+
+		<fieldset className="block-editor-responsive-block-control">
+			<legend className="block-editor-responsive-block-control__title">{ title }</legend>
+
+			<div className="block-editor-responsive-block-control__inner">
+				<ToggleControl
+					className="block-editor-responsive-block-control__toggle"
+					label={ toggleControlLabel }
+					checked={ ! isResponsive }
+					onChange={ onIsResponsiveChange }
+					help={ toggleHelpText }
+				/>
+
+				{ ! isResponsive && (
+					<div className="block-editor-responsive-block-control__group block-editor-responsive-block-control__group--default" >
```

Comment:
Same note here about BEM convention as at: https://github.com/WordPress/gutenberg/pull/17846#discussion_r368140643

Also:

- These classes are only referenced in tests. It's not strictly problematic to do so, but ideally the runtime implementation needn't be conformed specifically to tailor to tests.
- I think in representing this as some sort of on/off state (it either _is_ or _isn't_ "responsive"), we'd not need to refer to this explicitly as "default" but rather as the absence of "responsive". In other words, I'd suggest using modifier class `is-responsive` for the other condition, and for this one, simply omitting it.

The second point would actually enable us to simplify this implementation, where currently there is code duplication for `block-editor-responsive-block-control__group` largely because of how this class is assigned. With the above changes, it could become a unified wrapper of:

```jsx
<div className={ classnames( 'block-editor-responsive-block-control__group', { 'is-responsive': isResponsive } ) } hidden={ ! isResponsive }>
```